File: typescript/src/tests/TimeDepositCalculator.test.ts
import { TimeDeposit } from '../TimeDeposit'
import { TimeDepositCalculator } from '../TimeDepositCalculator'


test('Should update balance', () => {
  const plans: TimeDeposit[] = [new TimeDeposit(1, 'basic', 1234567.0, 45)]
  const calc = new TimeDepositCalculator()
  calc.updateBalance(plans)

  expect(1).toBe(1)
})

File: java/src/test/java/org/ikigaidigital/service/TimeDepositCalculatorTest.java
package org.ikigaidigital;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class TimeDepositCalculatorTest {
    @Test
    public void updateBalance_Test() {
        TimeDepositCalculator calc = new TimeDepositCalculator();
        TimeDeposit basic = new TimeDeposit(1, "basic", 1234567.00, 45);

        List<TimeDeposit> plans = List.of(basic);
        calc.updateBalance(plans);

        assertThat(basic.getBalance()).isEqualTo(1235595.81);
    }
}
